app/models/service_orchestration.rb

```diff
@@ -83,6 +83,10 @@ def post_provision_configure
     assign_vms_owner
   end
 
+  def my_zone
+    service_template.try(:my_zone)
```

Comment:
As discussed the problem with using `service_template` is that it can be deleted or changed to reference a different provider from when the service was deployed.  Better to reference the provider from the service_resources relationship to lookup the zone and use `super` if that is `nil`.